File: app/javascript/channels/test_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("TestChannel", {
  connected() {
    console.log("Connected to TestChannel")
    this.perform("ping", { test: "data" })
  },

  disconnected() {
    console.log("Disconnected from TestChannel")
  },

  received(data) {
    console.log("Received data from TestChannel:", data)
  }
})

File: app/channels/test_channel.rb
class TestChannel < ApplicationCable::Channel
  def subscribed
    stream_from "test_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def ping(data)
    ActionCable.server.broadcast("test_channel", { message: "pong", received: data })
  end
end
